File: Javascript/21_MergeTwoSortedLists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */

var mergeTwoLists = function (list1, list2) {
    let result = new ListNode();
    let listPosition = result;
    while (list1 && list2) {
        if (list1.val > list2.val) {
            listPosition.next = list2;
            list2 = list2.next;
        } else {
            listPosition.next = list1;
            list1 = list1.next;
        }
        listPosition = listPosition.next;
    }
    listPosition.next = (list1 ? list1 : list2);
    return result.next;
};

File: Ruby/21_merge_two_sorted_lists.rb
# Definition for singly-linked list.
class ListNode
    attr_accessor :val, :next
    def initialize(val = 0, _next = nil)
        @val = val
        @next = _next
    end
end
# @param {ListNode} list1
# @param {ListNode} list2
# @return {ListNode}
def merge_two_lists(list1, list2)
  list = ListNode.new(0)
  head = list
  while list1 && list2
    if list1.val <= list2.val
      list.next = list1
      list1 = list1.next
    else
      list.next = list2
      list2 = list2.next
    end
  end
  list.next = list1 || list2
  head.next
end

merge_two_lists([1,2,4], [1,3,4])
